server/src/device/types.ts

```diff
@@ -0,0 +1,15 @@
+import { IsNotEmpty } from 'class-validator';
+
+export interface Location {
+  latitude: number;
+  longitude: number;
+}
+
+export class SyncDeviceDTO {
+  @IsNotEmpty()
+  token: string;
+  @IsNotEmpty()
+  location: Location;
+
+  time?: string; // ISO 8601 format
```

Comment:
Don't we need an annotation here? There's an annotation @IsOptional(), I think it should be added here

Reply:
Fixing it!